Update compat data for PromiseRejectionEvent

TrevorBurnham opened the pull request:
Follow-up for #2100

Currently, the MDN docs indicate that `PromiseRejectionEvent` and the associated `window.onunhandledrejection` and `window.onrejectionhandled` properties are Chrome-only. I believe the CanIUse data for these events are correct: https://caniuse.com/#feat=unhandledrejection

Also, I believe this same data should appear on https://developer.mozilla.org/en-US/docs/Web/Events/unhandledrejection#Browser_compatibility.

ddbeck commented on api/WindowEventHandlers.json:
```diff
@@ -642,39 +659,56 @@
+              "version_added": null
             },
             "firefox": {
               "version_added": false,
```
I guess we should set this to version 55 and `"partial_implementation": true`: https://bugzilla.mozilla.org/show_bug.cgi?id=1338059

ddbeck commented on api/WindowEventHandlers.json:
```diff
@@ -642,39 +659,56 @@
+              "notes": "Firefox doesn't yet send the <a href='https://developer.mozilla.org/docs/Web/Events/unhandledrejection'>unhandledrejection</a> or understand <a href='https://developer.mozilla.org/docs/Web/Events/rejectionhandled'>rejectionhandled<a> events. Completing the implementation will be addressed in <a href='https://bugzil.la/1362272'>bug 1362272</a>."
             },
             "firefox_android": {
               "version_added": false,
```
Again, we should set this to version 55 and `"partial_implementation": true`

ddbeck commented on api/WindowEventHandlers.json:
```diff
@@ -642,39 +659,56 @@
             },
             "webview_android": {
-              "version_added": false
+              "version_added": null
```
Again, version 49 here

ddbeck commented on api/PromiseRejectionEvent.json:
```diff
@@ -42,16 +42,16 @@
           },
           "safari_ios": {
-            "version_added": false
+            "version_added": "11.4"
```
On the basis of https://bugs.webkit.org/show_bug.cgi?id=150358 I trust that these Safari changes actually shipped, but I couldn't place the version numbers and 11.4 isn't in our Safari for iOS data at all. I'm willing to let the desktop version be, but I'm inclined to suggest that we go with `null` for iOS, if we don't have a more direct source for this version number. `null` would also satisfy the linter.

vinyldarkscratch replied:
Based upon my research in #2006, I found that iOS 11.3 and 11.4 use Safari 11.1.  Since iOS 11.3 is the version where Safari is updated, we should set this one to "11.3".  At least, once #2006 has been completed.  :wink:

TrevorBurnham commented:
@ddbeck OK, I think I've addressed all of your feedback.

Regarding `safari_ios`: Looking at iOS devices in BrowserStack, I can confirm that `onunhandledrejection` is supported in iOS 12.0.0. Support was likely introduced in an earlier iOS 11.x release, but I can't pin that version down. Would it be better to set `"version_added": "12.0.0"` instead of `"version_added": null`?

TrevorBurnham commented:
> FYI, caniuse and Chrome Status are fine as sources if you don't have anything better. Be aware that they are second-hand sources.

OK, them I'm going to go with 11.4 as the `safari_ios` version, per caniuse. I've added that version to `safari_ios.json` to satisfy the linter: 0f84926

ddbeck commented:
@Elchi3 would you mind reviewing the browser change here? I don't feel confident about reviewing `safari_ios.json` additions myself

Elchi3 commented:
hm, I have no idea whether or not these are versions that we would care about, or just minor updates without feature changes. Does anyone have Safari release notes handy? We need to find out which of the 11.x releases are major releases with web platform feature changes.

Caniuse seems to have some grouping:
"11.0-11.2"
"11.3-11.4"
For us this would mean we would add only add 11.0 and 11.3, but we've already added 11.1...

ddbeck commented:
This is also related to #2006.

TrevorBurnham commented:
@vinyldarkscratch Per your comment, I've changed the iOS version specified here from `"11.4"` to to `"11.3"` and rebased against `master`.

Let me know if any other changes are needed here.

vinyldarkscratch commented:
Awesome, thanks for the update, and thank you for the updates!  Since this no longer makes any changes to the browsers, I think that this is ready.

@ddbeck Since you know much more about this PR than I do (and the PR looks great to me), want to give it one final review?
